lib/smart_proxy_dynflow_core/log.rb

```diff
@@ -40,6 +40,25 @@ def log_file
       end
     end
 
+    def initialize(file, *rest)
+      @fd = file.kind_of?(IO) ? file : File.open(file, 'a')
+      super(@fd, rest)
+    end
+
+    def add(*args)
+      handle_log_rolling if @roll_log
+      super(*args)
+    end
+
+    def roll_log
+      @roll_log = true
+    end
+
+    def handle_log_rolling
+      @roll_log = false
+      @fd.reopen unless log_file.kind_of? IO
```

Comment:
[log.rb#L34](https://github.com/adamruzicka/smart_proxy_dynflow/blob/99a8eb97c9e054c07577d9df79ea6ff9ba56ffad/lib/smart_proxy_dynflow_core/log.rb#L34)

There are two possibilities either it is a string which we treat as a filesystem path or it is stdout. Reopening stdout on logrotate doesn't make much sense